.ci/docker/ubuntu.Dockerfile

```diff
@@ -30,5 +30,6 @@ ENV CC="$cc" CXX="$cxx" CMAKE_GENERATOR="Ninja" CMAKE_EXPORT_COMPILE_COMMANDS=on
 RUN ls -lR src
 RUN cmake -B build -S . "$cmake_args"
 RUN cmake --build build --verbose
+RUN cmake --build build --target all_verify_interface_header_sets
```

Comment:
Not a big deal, but is the `--verbose` flag worth adding for the `all_verify_interface_header_sets` command? If we see a compiler error in a CI log, do we see enough detail to identify the relevant compilation commands?

Reply:
Sure I can add the verbose flag. Logs are free.

Reply:
https://github.com/beman-project/exemplar/pull/6 tries to remove the Dockerfiles. Maybe we should decide if we'll do that or not, as it may influence current PR.

Reply:
I have not checked failure case but I don't think verbose provide any useful information, should we still add the verbose flag?

Log I received from running with verbose flag locally:
```
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -S/home/bradwu/Desktop/exemplar -B/home/bradwu/Desktop/exemplar/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/gmake  -f CMakeFiles/Makefile2 all_verify_interface_header_sets
gmake[1]: Entering directory '/home/bradwu/Desktop/exemplar/build'
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -S/home/bradwu/Desktop/exemplar -B/home/bradwu/Desktop/exemplar/build --check-build-system CMakeFiles/Makefile.cmake 0
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_progress_start /home/bradwu/Desktop/exemplar/build/CMakeFiles 3
/usr/bin/gmake  -f CMakeFiles/Makefile2 CMakeFiles/all_verify_interface_header_sets.dir/all
gmake[2]: Entering directory '/home/bradwu/Desktop/exemplar/build'
/usr/bin/gmake  -f src/beman/exemplar/CMakeFiles/beman.exemplar.dir/build.make src/beman/exemplar/CMakeFiles/beman.exemplar.dir/depend
gmake[3]: Entering directory '/home/bradwu/Desktop/exemplar/build'
cd /home/bradwu/Desktop/exemplar/build && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_depends "Unix Makefiles" /home/bradwu/Desktop/exemplar /home/bradwu/Desktop/exemplar/src/beman/exemplar /home/bradwu/Desktop/exemplar/build /home/bradwu/Desktop/exemplar/build/src/beman/exemplar /home/bradwu/Desktop/exemplar/build/src/beman/exemplar/CMakeFiles/beman.exemplar.dir/DependInfo.cmake --color=
gmake[3]: Leaving directory '/home/bradwu/Desktop/exemplar/build'
/usr/bin/gmake  -f src/beman/exemplar/CMakeFiles/beman.exemplar.dir/build.make src/beman/exemplar/CMakeFiles/beman.exemplar.dir/build
gmake[3]: Entering directory '/home/bradwu/Desktop/exemplar/build'
[ 33%] Building CXX object src/beman/exemplar/CMakeFiles/beman.exemplar.dir/identity.cpp.o
cd /home/bradwu/Desktop/exemplar/build/src/beman/exemplar && /usr/bin/c++  -I/home/bradwu/Desktop/exemplar/src/beman/exemplar/../../../include -std=gnu++20 -MD -MT src/beman/exemplar/CMakeFiles/beman.exemplar.dir/identity.cpp.o -MF CMakeFiles/beman.exemplar.dir/identity.cpp.o.d -o CMakeFiles/beman.exemplar.dir/identity.cpp.o -c /home/bradwu/Desktop/exemplar/src/beman/exemplar/identity.cpp
[ 66%] Linking CXX static library libbeman.exemplar.a
cd /home/bradwu/Desktop/exemplar/build/src/beman/exemplar && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -P CMakeFiles/beman.exemplar.dir/cmake_clean_target.cmake
cd /home/bradwu/Desktop/exemplar/build/src/beman/exemplar && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_link_script CMakeFiles/beman.exemplar.dir/link.txt --verbose=1
/usr/bin/ar qc libbeman.exemplar.a CMakeFiles/beman.exemplar.dir/identity.cpp.o
/usr/bin/ranlib libbeman.exemplar.a
gmake[3]: Leaving directory '/home/bradwu/Desktop/exemplar/build'
[ 66%] Built target beman.exemplar
/usr/bin/gmake  -f src/beman/exemplar/CMakeFiles/beman.exemplar_verify_interface_header_sets.dir/build.make src/beman/exemplar/CMakeFiles/beman.exemplar_verify_interface_header_sets.dir/depend
gmake[3]: Entering directory '/home/bradwu/Desktop/exemplar/build'
cd /home/bradwu/Desktop/exemplar/build && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_depends "Unix Makefiles" /home/bradwu/Desktop/exemplar /home/bradwu/Desktop/exemplar/src/beman/exemplar /home/bradwu/Desktop/exemplar/build /home/bradwu/Desktop/exemplar/build/src/beman/exemplar /home/bradwu/Desktop/exemplar/build/src/beman/exemplar/CMakeFiles/beman.exemplar_verify_interface_header_sets.dir/DependInfo.cmake --color=
gmake[3]: Leaving directory '/home/bradwu/Desktop/exemplar/build'
/usr/bin/gmake  -f src/beman/exemplar/CMakeFiles/beman.exemplar_verify_interface_header_sets.dir/build.make src/beman/exemplar/CMakeFiles/beman.exemplar_verify_interface_header_sets.dir/build
gmake[3]: Entering directory '/home/bradwu/Desktop/exemplar/build'
[100%] Building CXX object src/beman/exemplar/CMakeFiles/beman.exemplar_verify_interface_header_sets.dir/beman.exemplar_verify_interface_header_sets/beman/exemplar/identity.hpp.cxx.o
cd /home/bradwu/Desktop/exemplar/build/src/beman/exemplar && /usr/bin/c++  -I/home/bradwu/Desktop/exemplar/src/beman/exemplar/../../../include -x c++ -std=gnu++20 -MD -MT src/beman/exemplar/CMakeFiles/beman.exemplar_verify_interface_header_sets.dir/beman.exemplar_verify_interface_header_sets/beman/exemplar/identity.hpp.cxx.o -MF CMakeFiles/beman.exemplar_verify_interface_header_sets.dir/beman.exemplar_verify_interface_header_sets/beman/exemplar/identity.hpp.cxx.o.d -o CMakeFiles/beman.exemplar_verify_interface_header_sets.dir/beman.exemplar_verify_interface_header_sets/beman/exemplar/identity.hpp.cxx.o -c /home/bradwu/Desktop/exemplar/build/src/beman/exemplar/beman.exemplar_verify_interface_header_sets/beman/exemplar/identity.hpp.cxx
gmake[3]: Leaving directory '/home/bradwu/Desktop/exemplar/build'
[100%] Built target beman.exemplar_verify_interface_header_sets
/usr/bin/gmake  -f CMakeFiles/all_verify_interface_header_sets.dir/build.make CMakeFiles/all_verify_interface_header_sets.dir/depend
gmake[3]: Entering directory '/home/bradwu/Desktop/exemplar/build'
cd /home/bradwu/Desktop/exemplar/build && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_depends "Unix Makefiles" /home/bradwu/Desktop/exemplar /home/bradwu/Desktop/exemplar /home/bradwu/Desktop/exemplar/build /home/bradwu/Desktop/exemplar/build /home/bradwu/Desktop/exemplar/build/CMakeFiles/all_verify_interface_header_sets.dir/DependInfo.cmake --color=
gmake[3]: Leaving directory '/home/bradwu/Desktop/exemplar/build'
/usr/bin/gmake  -f CMakeFiles/all_verify_interface_header_sets.dir/build.make CMakeFiles/all_verify_interface_header_sets.dir/build
gmake[3]: Entering directory '/home/bradwu/Desktop/exemplar/build'
gmake[3]: Nothing to be done for 'CMakeFiles/all_verify_interface_header_sets.dir/build'.
gmake[3]: Leaving directory '/home/bradwu/Desktop/exemplar/build'
[100%] Built target all_verify_interface_header_sets
gmake[2]: Leaving directory '/home/bradwu/Desktop/exemplar/build'
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_progress_start /home/bradwu/Desktop/exemplar/build/CMakeFiles 0
gmake[1]: Leaving directory '/home/bradwu/Desktop/exemplar/build'
```

Reply:
I can work on this later too but can you help me produce a case where the `all_verify_interface_header_sets` fails? I can go check the log and see if verbose provide more infomation.

Reply:
For instance, I would expect the following to work in a normally-formed CMake library but fail an `all_verify_interface_header_sets` check.

```
// @file something.hpp
struct something::Widget {
  // BUG: Uses std::vector but doesn't include <vector>
  std::vector vec;
};
```

```
// @file something.cpp
#include <vector>
// something.hpp as expanded here gets vector defined for "free"
// by the preceeding #include, so something.cpp compiles fine
#include <something.hpp>

inline void something::foobar(Widget const& widget) {
    return widget.vec.size();
}
```

I just want to make sure the person looking at a failure log can tell:

* That `something.hpp` is being checked with a specific compile command
  * (Something about building a temporary like `$some_build_dir/something.hpp.cpp` probably)
* What the compile command is, because flags like header search paths can be relevant
* Exactly what the problem is

If we get that with `--verbose` off, I'm happy. We can land a shorter log and everyone wins.

If we don't get it regardless of whether `--verbose` is on or off, let's make an upstream CMake issue. I wouldn't expect tweaking this would be tricky or expensive for them.

Reply:
> > #6 tries to remove the Dockerfiles. Maybe we should decide if we'll do that or not, as it may influence current PR.
> 
> Since #6 is closed, no worries on this PR, right?

We shouldn't worry about that.

Reply:
<details>
<summary>Without verbose Flag</summary>

```
⋊> ~/D/test-cmake cmake --build build --target verify_interface_header_sets
[ 40%] Built target something_lib
[ 60%] Generating verify_something.hpp.cpp

[ 60%] Generating verify_something.hpp.cpp

[ 80%] Building CXX object CMakeFiles/verify_something.dir/verify_something.hpp.cpp.o
cc1plus: fatal error: /home/bradwu/Desktop/test-cmake/build/verify_something.hpp.cpp: No such file or directory
compilation terminated.
gmake[3]: *** [CMakeFiles/verify_something.dir/build.make:80: CMakeFiles/verify_something.dir/verify_something.hpp.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:113: CMakeFiles/verify_something.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:146: CMakeFiles/verify_interface_header_sets.dir/rule] Error 2
gmake: *** [Makefile:150: verify_interface_header_sets] Error 2
```

</details>

<details>
<summary>With Verbose Flag</summary>

```
⋊> ~/D/test-cmake cmake --build build --target verify_interface_header_sets --verbose
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -S/home/bradwu/Desktop/test-cmake -B/home/bradwu/Desktop/test-cmake/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/gmake  -f CMakeFiles/Makefile2 verify_interface_header_sets
gmake[1]: Entering directory '/home/bradwu/Desktop/test-cmake/build'
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -S/home/bradwu/Desktop/test-cmake -B/home/bradwu/Desktop/test-cmake/build --check-build-system CMakeFiles/Makefile.cmake 0
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_progress_start /home/bradwu/Desktop/test-cmake/build/CMakeFiles 5
/usr/bin/gmake  -f CMakeFiles/Makefile2 CMakeFiles/verify_interface_header_sets.dir/all
gmake[2]: Entering directory '/home/bradwu/Desktop/test-cmake/build'
/usr/bin/gmake  -f CMakeFiles/something_lib.dir/build.make CMakeFiles/something_lib.dir/depend
gmake[3]: Entering directory '/home/bradwu/Desktop/test-cmake/build'
cd /home/bradwu/Desktop/test-cmake/build && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_depends "Unix Makefiles" /home/bradwu/Desktop/test-cmake /home/bradwu/Desktop/test-cmake /home/bradwu/Desktop/test-cmake/build /home/bradwu/Desktop/test-cmake/build /home/bradwu/Desktop/test-cmake/build/CMakeFiles/something_lib.dir/DependInfo.cmake --color=
gmake[3]: Leaving directory '/home/bradwu/Desktop/test-cmake/build'
/usr/bin/gmake  -f CMakeFiles/something_lib.dir/build.make CMakeFiles/something_lib.dir/build
gmake[3]: Entering directory '/home/bradwu/Desktop/test-cmake/build'
gmake[3]: Nothing to be done for 'CMakeFiles/something_lib.dir/build'.
gmake[3]: Leaving directory '/home/bradwu/Desktop/test-cmake/build'
[ 40%] Built target something_lib
/usr/bin/gmake  -f CMakeFiles/verify_something.dir/build.make CMakeFiles/verify_something.dir/depend
gmake[3]: Entering directory '/home/bradwu/Desktop/test-cmake/build'
[ 60%] Generating verify_something.hpp.cpp
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -E echo #include\ <something.hpp> > /home/bradwu/Desktop/test-cmake/build/verify_something.hpp.cpp

cd /home/bradwu/Desktop/test-cmake/build && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_depends "Unix Makefiles" /home/bradwu/Desktop/test-cmake /home/bradwu/Desktop/test-cmake /home/bradwu/Desktop/test-cmake/build /home/bradwu/Desktop/test-cmake/build /home/bradwu/Desktop/test-cmake/build/CMakeFiles/verify_something.dir/DependInfo.cmake --color=
gmake[3]: Leaving directory '/home/bradwu/Desktop/test-cmake/build'
/usr/bin/gmake  -f CMakeFiles/verify_something.dir/build.make CMakeFiles/verify_something.dir/build
gmake[3]: Entering directory '/home/bradwu/Desktop/test-cmake/build'
[ 60%] Generating verify_something.hpp.cpp
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.3/bin/cmake -E echo #include\ <something.hpp> > /home/bradwu/Desktop/test-cmake/build/verify_something.hpp.cpp

[ 80%] Building CXX object CMakeFiles/verify_something.dir/verify_something.hpp.cpp.o
/usr/bin/c++  -I/home/bradwu/Desktop/test-cmake -std=gnu++17 -MD -MT CMakeFiles/verify_something.dir/verify_something.hpp.cpp.o -MF CMakeFiles/verify_something.dir/verify_something.hpp.cpp.o.d -o CMakeFiles/verify_something.dir/verify_something.hpp.cpp.o -c /home/bradwu/Desktop/test-cmake/build/verify_something.hpp.cpp
cc1plus: fatal error: /home/bradwu/Desktop/test-cmake/build/verify_something.hpp.cpp: No such file or directory
compilation terminated.
gmake[3]: *** [CMakeFiles/verify_something.dir/build.make:80: CMakeFiles/verify_something.dir/verify_something.hpp.cpp.o] Error 1
gmake[3]: Leaving directory '/home/bradwu/Desktop/test-cmake/build'
gmake[2]: *** [CMakeFiles/Makefile2:113: CMakeFiles/verify_something.dir/all] Error 2
gmake[2]: Leaving directory '/home/bradwu/Desktop/test-cmake/build'
gmake[1]: *** [CMakeFiles/Makefile2:146: CMakeFiles/verify_interface_header_sets.dir/rule] Error 2
gmake[1]: Leaving directory '/home/bradwu/Desktop/test-cmake/build'
gmake: *** [Makefile:150: verify_interface_header_sets] Error 2
```

</details>

Verbose doesn't provide more useful information, and honestly the error log is pretty bad. The error fails because it cannot find the `verify_something.hpp.cpp`, even with verbose on nothing indicate the reason for failure to generate this file.

Tested on `cmake version 3.26.3`

Reply:
I tested on `cmake version 3.30.3`, same output, I will file a issue upstream, let's omit the verbose flag.

<details>
<summary>version 3.30.3 with verbose log</summary>

```
⋊> ~/D/test-cmake cmake --build build --verbose                                                                                                                                (base) 13:45:00
Change Dir: '/home/bradwu/Desktop/test-cmake/build'

Run Build Command(s): /home/linuxbrew/.linuxbrew/Cellar/cmake/3.30.3/bin/cmake -E env VERBOSE=1 /usr/bin/gmake -f Makefile
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.30.3/bin/cmake -S/home/bradwu/Desktop/test-cmake -B/home/bradwu/Desktop/test-cmake/build --check-build-system CMakeFiles/Makefile.cmake 0
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.30.3/bin/cmake -E cmake_progress_start /home/bradwu/Desktop/test-cmake/build/CMakeFiles /home/bradwu/Desktop/test-cmake/build//CMakeFiles/progress.marks
/usr/bin/gmake  -f CMakeFiles/Makefile2 all
gmake[1]: Entering directory '/home/bradwu/Desktop/test-cmake/build'
/usr/bin/gmake  -f CMakeFiles/something_lib.dir/build.make CMakeFiles/something_lib.dir/depend
gmake[2]: Entering directory '/home/bradwu/Desktop/test-cmake/build'
cd /home/bradwu/Desktop/test-cmake/build && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.30.3/bin/cmake -E cmake_depends "Unix Makefiles" /home/bradwu/Desktop/test-cmake /home/bradwu/Desktop/test-cmake /home/bradwu/Desktop/test-cmake/build /home/bradwu/Desktop/test-cmake/build /home/bradwu/Desktop/test-cmake/build/CMakeFiles/something_lib.dir/DependInfo.cmake "--color="
Dependencies file "CMakeFiles/something_lib.dir/something.cpp.o.d" is newer than depends file "/home/bradwu/Desktop/test-cmake/build/CMakeFiles/something_lib.dir/compiler_depend.internal".
Consolidate compiler generated dependencies of target something_lib
gmake[2]: Leaving directory '/home/bradwu/Desktop/test-cmake/build'
/usr/bin/gmake  -f CMakeFiles/something_lib.dir/build.make CMakeFiles/something_lib.dir/build
gmake[2]: Entering directory '/home/bradwu/Desktop/test-cmake/build'
gmake[2]: Nothing to be done for 'CMakeFiles/something_lib.dir/build'.
gmake[2]: Leaving directory '/home/bradwu/Desktop/test-cmake/build'
[ 40%] Built target something_lib
/usr/bin/gmake  -f CMakeFiles/verify_something.dir/build.make CMakeFiles/verify_something.dir/depend
gmake[2]: Entering directory '/home/bradwu/Desktop/test-cmake/build'
[ 60%] Generating verify_something.hpp.cpp
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.30.3/bin/cmake -E echo #include\ <something.hpp> > /home/bradwu/Desktop/test-cmake/build/verify_something.hpp.cpp

cd /home/bradwu/Desktop/test-cmake/build && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.30.3/bin/cmake -E cmake_depends "Unix Makefiles" /home/bradwu/Desktop/test-cmake /home/bradwu/Desktop/test-cmake /home/bradwu/Desktop/test-cmake/build /home/bradwu/Desktop/test-cmake/build /home/bradwu/Desktop/test-cmake/build/CMakeFiles/verify_something.dir/DependInfo.cmake "--color="
gmake[2]: Leaving directory '/home/bradwu/Desktop/test-cmake/build'
/usr/bin/gmake  -f CMakeFiles/verify_something.dir/build.make CMakeFiles/verify_something.dir/build
gmake[2]: Entering directory '/home/bradwu/Desktop/test-cmake/build'
[ 60%] Generating verify_something.hpp.cpp
/home/linuxbrew/.linuxbrew/Cellar/cmake/3.30.3/bin/cmake -E echo #include\ <something.hpp> > /home/bradwu/Desktop/test-cmake/build/verify_something.hpp.cpp

[ 80%] Building CXX object CMakeFiles/verify_something.dir/verify_something.hpp.cpp.o
/usr/bin/c++  -I/home/bradwu/Desktop/test-cmake -std=gnu++17 -MD -MT CMakeFiles/verify_something.dir/verify_something.hpp.cpp.o -MF CMakeFiles/verify_something.dir/verify_something.hpp.cpp.o.d -o CMakeFiles/verify_something.dir/verify_something.hpp.cpp.o -c /home/bradwu/Desktop/test-cmake/build/verify_something.hpp.cpp
cc1plus: fatal error: /home/bradwu/Desktop/test-cmake/build/verify_something.hpp.cpp: No such file or directory
compilation terminated.
gmake[2]: *** [CMakeFiles/verify_something.dir/build.make:80: CMakeFiles/verify_something.dir/verify_something.hpp.cpp.o] Error 1
gmake[2]: Leaving directory '/home/bradwu/Desktop/test-cmake/build'
gmake[1]: *** [CMakeFiles/Makefile2:113: CMakeFiles/verify_something.dir/all] Error 2
gmake[1]: Leaving directory '/home/bradwu/Desktop/test-cmake/build'
gmake: *** [Makefile:91: all] Error 2
```
</details>